addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/handler/DSMRMeterHandler.java

```diff
@@ -0,0 +1,205 @@
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dsmr.handler;
+
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.dsmr.internal.device.cosem.CosemObject;
+import org.openhab.binding.dsmr.internal.device.cosem.CosemValue;
+import org.openhab.binding.dsmr.internal.device.cosem.CosemValueDescriptor;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeter;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeterConfiguration;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeterConstants;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeterDescriptor;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeterListener;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeterType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The MeterHandler will create logic DSMR meter ThingTypes
+ */
+public class DSMRMeterHandler extends BaseThingHandler implements DSMRMeterListener {
+    // logger
+    private final Logger logger = LoggerFactory.getLogger(DSMRMeterHandler.class);
+
+    // The DSMRMeter instance
+    private DSMRMeter meter = null;
+
+    // Timestamp when last values were received
+    private long lastValuesReceivedTs = 0;
+
+    // Reference to the meter watchdog
+    private ScheduledFuture<?> meterWatchdog;
+
+    /**
+     * Creates a new MeterHandler for the given Thing
+     *
+     * @param thing {@link Thing} to create the MeterHandler for
+     */
+    public DSMRMeterHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * DSMR Meter don't support handling commands
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No comments can be handled
+    }
+
+    /**
+     * Initializes a DSMR Meter
+     *
+     * This method will load the corresponding configuration
+     */
+    @Override
+    public void initialize() {
+        logger.debug("Initialize MeterHandler for Thing {}", getThing().getUID());
+
+        Configuration config = getThing().getConfiguration();
+        DSMRMeterConfiguration meterConfig = null;
+        DSMRMeterDescriptor meterDescriptor = null;
+
+        if (config != null) {
```

Comment:
No i did not ran into troubels. I agree about the documentation and removed the unecessary check